src/ClientHandle.cpp

```diff
@@ -3350,7 +3350,7 @@ void cClientHandle::SetSelf(cClientHandlePtr a_Self)
 bool cClientHandle::SetState(eState a_NewState)
 {
 	cCSLock Lock(m_CSState);
-	if (a_NewState < m_State)
+	if (a_NewState <= m_State)
```

Comment:
`SetState(csDestroying)` should only succeed once, so need to return false on `a_NewState == m_State`.